src/TaskModel.vala

```diff
@@ -66,6 +65,28 @@ public class Tasks.TaskModel : Object {
     private void create_task_list_client (E.Source task_list) {
         try {
             var client = (ECal.Client) ECal.Client.connect_sync (task_list, ECal.ClientSourceType.TASKS, -1, null);
+
+            get_registry.begin ((obj, res) => {
+                try {
+                    var registry = get_registry.end (res);
+                    if (get_collection_backend_name (task_list, registry) == "google") {
+                        var refresh = (E.SourceRefresh) task_list.get_extension (E.SOURCE_EXTENSION_REFRESH);
+                        Timeout.add_seconds (refresh.interval_minutes * 60 - 1, () => {
+                            if (!refresh.enabled) {
+                                return Source.REMOVE;
+                            }
+
+                            update_gtasks_tasklist.begin (task_list, client, registry);
+                            return Source.CONTINUE;
+                        });
+
+                        update_gtasks_tasklist.begin (task_list, client, registry);
+                    }
+                } catch (Error e) {
+                    warning ("cant get registry: %s", e.message);
+                }
+            });
```

Comment:
Not sure what happened, didn't we agree on solving refresh in another PR?

Reply:
Yeah, I think the merge take the commit back.